airbyte-webapp/src/components/connectorBuilder/Builder/PaginationSection.tsx

```diff
@@ -0,0 +1,180 @@
+import { useField } from "formik";
+
+import GroupControls from "components/GroupControls";
+import { ControlLabels } from "components/LabeledControl";
+
+import { injectIntoValues } from "../types";
+import { BuilderCard } from "./BuilderCard";
+import { BuilderField } from "./BuilderField";
+import { BuilderOneOf } from "./BuilderOneOf";
+
+interface PaginationSectionProps {
+  streamFieldPath: (fieldPath: string) => string;
+}
+
+export const PaginationSection: React.FC<PaginationSectionProps> = ({ streamFieldPath }) => {
+  const [field, , helpers] = useField(streamFieldPath("paginator"));
+
+  const handleToggle = (newToggleValue: boolean) => {
+    if (newToggleValue) {
+      helpers.setValue({
+        strategy: {
+          type: "OffsetIncrement",
+        },
+        pageSizeOption: {
+          inject_into: "request_parameter",
+        },
+        pageTokenOption: {
+          inject_into: "request_parameter",
+        },
+      });
+    } else {
+      helpers.setValue(undefined);
+    }
+  };
+  const toggledOn = field.value !== undefined;
+
+  const pageTokenOption = (
+    <GroupControls
+      label={
+        <ControlLabels
+          label="Page token option"
+          infoTooltipContent="Configures how the page token will be sent in requests to the source API"
+        />
+      }
+    >
+      <BuilderField
+        type="enum"
+        path={streamFieldPath("paginator.pageTokenOption.inject_into")}
+        options={injectIntoValues}
+        label="Inject into"
+        tooltip="Configures where the page token should be set on the HTTP requests"
+      />
+      <BuilderField
+        type="string"
+        path={streamFieldPath("paginator.pageTokenOption.field_name")}
+        label="Field name"
+        tooltip="Configures which key should be used in the location that the page token is being injected into"
+        optional
+      />
+    </GroupControls>
+  );
+
+  const pageSizeOption = (
```

Comment:
A similar situation here - you need to set the page size if yopu want to use the page size option:
<img width="1030" alt="Screenshot 2022-12-21 at 14 44 47" src="https://user-images.githubusercontent.com/1508364/208919679-21817c0b-75ea-4f27-b59c-72c7be9e46e7.png">

Reply:
I think putting the request options behind a checkbox like you suggested and looking for those to make page size optional or required would work.